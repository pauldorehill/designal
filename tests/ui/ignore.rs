use designal::Designal;

#[derive(Designal)]
#[designal(trim_start = "Human")]
#[designal(ignore)]
struct HumanBean();

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean1 {
    #[designal(ignore)]
    #[designal(ignore)]
    taste: String,
}
#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean2 {
    #[designal(ignore)]
    #[designal(remove)]
    taste: String,
}

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean3 {
    #[designal(ignore)]
    #[designal(rename = "crunch")]
    taste: String,
}

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean4 {
    #[designal(ignore)]
    #[designal(keep_rc)]
    taste: String,
}

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean5 {
    #[designal(ignore)]
    #[designal(keep_arc)]
    taste: String,
}

fn main() {}
