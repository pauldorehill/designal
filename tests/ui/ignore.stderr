error: Ignore is not valid at the container level
 --> $DIR/ignore.rs:5:12
  |
5 | #[designal(ignore)]
  |            ^^^^^^

error: You should only ignore once
  --> $DIR/ignore.rs:12:16
   |
12 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:18:16
   |
18 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:26:16
   |
26 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:34:16
   |
34 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:42:16
   |
42 |     #[designal(ignore)]
   |                ^^^^^^
